tokio/src/runtime/id.rs

```diff
@@ -32,18 +32,14 @@ use std::fmt;
 #[derive(Clone, Copy, Debug, Hash, Eq, PartialEq)]
 pub struct Id(u64);
```

Comment:
You could make the id into a `NonZeroU64` since we make sure to not use zero as an id.

Reply:
What's the benefit of using `NonZeroU64` here given that it's not public and is always instanciated from a `u64`?

I'm not against making the change, I just don't understand the benefit (and I know it's going to need an `unwrap()`).

Reply:
It makes an `Option<Id>` smaller.

You don't necessarily need an unwrap if you change `src/runtime/task/list.rs` to the following:
```Rust
fn get_next_id() -> NonZeroU64 {
    loop {
        let id = NEXT_OWNED_TASKS_ID.fetch_add(1, Ordering::Relaxed);
        if let Some(id) = NonZeroU64::new(id) {
            return id;
        }
    }
}
```

Reply:
Thanks for the answer. I get it now, and I hadn't thought of that loop either!